pytorch_lightning/trainer/trainer.py

```diff
@@ -382,10 +382,19 @@ def __init__(
         self.slurm_connector = SLURMConnector(self)
         self.tuner = Tuner(self)
 
-        # max_epochs won't default to 1000 if max_steps/max_time are specified (including being set to -1).
+        if max_epochs is None:
+            # max_epochs won't default to 1000 if max_steps/max_time are non-default values.
+            max_epochs = 1000 if (max_steps == -1 and max_time is None) else -1
+
+        elif max_epochs < -1:
+            # Allow max_epochs to be zero, since this will be handled by fit_loop.done
+            raise MisconfigurationException(
+                f"`max_epochs` must be a positive integer or -1. You passed in {max_epochs}."
+            )
```

Comment:
let's keep this in the fit loop

Reply:
this was marked as resolved but doesn't look like it - this check is already in the `FitLoop.__init__`

Reply:
Whoops sorry I thought it got added to my last commit. Looked like its still staged